### PR TITLE
feat(filter-side-panel): Add filter-side-panel and package patternfly-react-extensions

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
affects: patternfly-react-extensions

**What**:
Adds the first extension component - FilterSidePanel

**Link to Storybook**:
https://jeff-phillips-18.github.io/patternfly-react/

@tlwu2013 @serenamarie125

### CODE REVIEW

**Comment by dtaylor113** [comment]
Hi @jeff-phillips-18, storybook link not working

**Comment by dtaylor113** [comment]
Hi @jeff-phillips-18, why does FilterSidePanel need to be a 'extension component' as opposed to a normal component?

File: packages/patternfly-3/patternfly-react-extensions/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# @patternfly/react-extensions
```
Comment (jschuler) [inline packages/patternfly-3/patternfly-react-extensions/README.md]: Do we have a pattern for pf3 vs pf4 package naming? I am wondering if we have a similar extensions package for pf4-react what the name should be.

Comment (jeff-phillips-18) [reply packages/patternfly-3/patternfly-react-extensions/README.md]: oops, I forgot to update this. It should be `patternfly-react-extensions`

File: packages/patternfly-3/patternfly-react-extensions/src/components/FilterSidePanel/FilterSidePanelCategoryItem.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+const FilterSidePanelCategoryItem = ({ children, className, icon, count, ...props }) => {
+  const classes = classNames('filter-panel-pf-category-item', className);
+
+  return (
+    <div className={classes}>
+      <Checkbox {...props}>
+        {icon && <span className="item-icon">{icon}</span>}
+        {children}
+        {Number.isInteger(count) && <span className="item-count">{`(${count})`}</span>}
```
Comment (jschuler) [inline packages/patternfly-3/patternfly-react-extensions/src/components/FilterSidePanel/FilterSidePanelCategoryItem.js]: Inside the span can just have `{count}`

Comment (jeff-phillips-18) [reply packages/patternfly-3/patternfly-react-extensions/src/components/FilterSidePanel/FilterSidePanelCategoryItem.js]: we want the count inside parens, like `(3)`

**Comment by jgiardino** [comment]
This looks great! 

If you have time, it would be worth investigating the use of [`<fieldset>` and `<legend>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) elements in this component to group the checkboxes together. That will require probably some additional styling, though. Here's an article from Léonie Watson about the a11y benefits: https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/

Also, for the checkboxes with stars, adding a span with a text label would help make the label more accessible. For example:
```
<label title="">
  <input type="checkbox">
  <span class="sr-only">5 star rating</span>
  <span class="item-icon">
    ...
```

The `<label>` elements have an empty `title` attribute, e.g. `<label title="">...</label>`

**Comment by jgiardino** [comment]
This looks great! 

If you have time, it would be worth investigating the use of [`<fieldset>` and `<legend>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) elements in this component to group the checkboxes together. That will require probably some additional styling, though. Here's an article from Léonie Watson about the a11y benefits: https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/

Also, for the checkboxes with stars, adding a span with a text label would help make the label more accessible. For example:
```
<label title="">
  <input type="checkbox">
  <span class="sr-only">5 star rating</span>
  <span class="item-icon">
    ...
```

The `<label>` elements have an empty `title` attribute, e.g. `<label title="">...</label>`

**Comment by jeff-phillips-18** [comment]
Address review comments. Thanks!

**Pull request merged**